examples/sock_tcp_echo/Makefile

```diff
@@ -0,0 +1,37 @@
+# name of your application
+APPLICATION = sock_tcp_echo
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= native
+
+# default to using GNRC
+LWIP ?= 0
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+USEMODULE += sock_tcp
+USEMODULE += netdev_default
+
+USEMODULE += shell
+USEMODULE += shell_commands
+USEMODULE += ps
+USEMODULE += netutils
+
+ifeq (1, $(LWIP))
+  USEMODULE += ipv6_addr
+  USEMODULE += lwip_ipv6_autoconfig
+  USEMODULE += lwip_netdev
+else
+  USEMODULE += auto_init_gnrc_netif
+  USEMODULE += gnrc_ipv6_default
+  # we want to be able to open two sockets
+  CFLAGS += -DCONFIG_GNRC_TCP_RCV_BUFFERS=2
```

Comment:
Do that in `app.gnrc.config`? You can, I believe then add it as a config file here. If GNRC TCP does not have Kconfig support yet, NVM (alternatively, a good opportunity to add some ;-).

Reply:
Of course GNRC TCP has Kconfig support ;)

Reply:
Do I just plop `CONFIG_GNRC_TCP_RCV_BUFFERS=2` into `app.gnrc.config` or is there something else to do?

Reply:
I think you also need to activate the module config `CONFIG_KCONFIG_USEMODULE_GNRC_TCP=y`, but Kconfig will complain if something goes wrong (that's the advantage of Kconfig instead of just using CFLAGS ;-))

Reply:
Since you are already tweaking the gnrc_tcp configuration, you can enable the faster connection teardown algorithm via:
GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN=1

Reply:
> GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN=1

CONFIG_..., right?

Reply:
> I think you also need to activate the module config `CONFIG_KCONFIG_USEMODULE_GNRC_TCP=y`, but Kconfig will complain if something goes wrong (that's the advantage of Kconfig instead of just using CFLAGS ;-))

I mean to actually make the build system use the app.config file. It just being present doesn't make the build system pick it up.

Reply:
/summon @leandrolanzieri

Reply:
> > I think you also need to activate the module config `CONFIG_KCONFIG_USEMODULE_GNRC_TCP=y`, but Kconfig will complain if something goes wrong (that's the advantage of Kconfig instead of just using CFLAGS ;-))
> 
> I mean to actually make the build system use the app.config file. It just being present doesn't make the build system pick it up.

My configuration file:
```makefile
CONFIG_KCONFIG_USEMODULE_GNRC_TCP=y
CONFIG_GNRC_TCP_RCV_BUFFERS=2
```
When placed as `app.config` it was applied to the end-configuration in `autoconf.h` (maybe try cleaning first).

The better option would be in the Makefile:
```makefile
ifeq (1, $(LWIP))
  USEMODULE += ipv6_addr
  USEMODULE += lwip_ipv6_autoconfig
  USEMODULE += lwip_netdev
else
  USEMODULE += auto_init_gnrc_netif
  USEMODULE += gnrc_ipv6_default
  KCONFIG_ADD_CONFIG += app.gnrc.config
endif
```

This alone would always run Kconfig, as there is an `.config` file in the app directory. To avoid this, add the following to the Makefile:
```makefile
# As there is a .config we want to explicitly disable Kconfig by setting
# the variable to empty
SHOULD_RUN_KCONFIG ?=
```

Reply:
Kconfig is on the way out, so we don't need this anymore.